### PR TITLE
GSoC ideas page

### PR DESCRIPTION
**Pull request by ocefpaf**
@CJ-Wright do you mind taking a quick look at this? It is mostly a transcript of https://github.com/regro/cf-scripts/milestone/4.

Please feel free to add/remove and edit stuff. We need this merge ASAP so I request conda-forge to be added to GSoC via NumFOCUS. (Sorry for the last time thing but I'm finally getting back on my feet, literally ;-p)

### CODE REVIEW

**Comment by CJ-Wright** [comment]
I would add the following issues as well, they might be a bit more involved:
https://github.com/regro/cf-scripts/issues/665
https://github.com/regro/cf-scripts/issues/664
https://github.com/regro/cf-scripts/issues/595
https://github.com/regro/cf-scripts/issues/663

we also have a [gsoc label on cf-scripts](https://github.com/regro/cf-scripts/labels/GSOC).

**Comment by ocefpaf** [comment]
> I would add the following issues as well

Perfect! Thanks!! Adding those ASAP.

> they might be a bit more involved:

BTW, how is my guessing about the difficulty levels? Also, is the text describing them OK?

File: _posts/2020-02-04-GSoC.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Here are some ideas towards that goal:
+
+1. Separate git logic from filtering and migration logic.
+
+   Currently the filtering and migration logic (which is common to regular and mini migrators) is coupled to the git/github logic via the class structure. We might consider separating this. This would make testing simpler, since we could test the git logic independently of the actual migration logic. It would also reduce the amount of code we keep.
+
+   Issue: https://github.com/regro/cf-scripts/issues/693
+   Experience Level: moderate
+
+2. Refactor `auto_tick.xsh`
```
Comment (CJ-Wright) [inline _posts/2020-02-04-GSoC.md]:
```suggestion
2. Refactor `auto_tick.py`
```

**Comment by ocefpaf** [comment]
Merging this so I can submit it to NumFOCUS. We can edit it more later.

**Pull request merged**